src/test/kotlin/io/snyk/plugin/ui/toolwindow/SnykAuthPanelTest.kt

```diff
@@ -0,0 +1,139 @@
+package io.snyk.plugin.ui.toolwindow
+
+import com.intellij.openapi.application.Application
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.components.service
+import com.intellij.openapi.project.Project
+import com.intellij.util.messages.MessageBus
+import io.mockk.every
+import io.mockk.justRun
+import io.mockk.mockk
+import io.mockk.mockkStatic
+import io.mockk.unmockkAll
+import io.mockk.verify
+import io.snyk.plugin.events.SnykSettingsListener.Companion.SNYK_SETTINGS_TOPIC
+import io.snyk.plugin.services.SnykAnalyticsService
+import io.snyk.plugin.services.SnykApplicationSettingsStateService
+import io.snyk.plugin.services.SnykCliAuthenticationService
+import io.snyk.plugin.services.download.SnykCliDownloaderService
+import junit.framework.TestCase.assertEquals
+import junit.framework.TestCase.assertNotNull
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import snyk.UIComponentFinder
+import snyk.amplitude.AmplitudeExperimentService
+
+class SnykAuthPanelTest {
+    private val project = mockk<Project>()
+    private val analyticsService = mockk<SnykAnalyticsService>()
+
+    @Before
+    fun setUp() {
+        unmockkAll()
+        mockkStatic(ApplicationManager::class)
+        val application = mockk<Application>()
+        every { ApplicationManager.getApplication() } returns application
+        val messageBus = mockk<MessageBus>(relaxed = true)
+        every { application.messageBus } returns messageBus
+        every { project.messageBus } returns messageBus
+        every { messageBus.syncPublisher(SNYK_SETTINGS_TOPIC) } returns mockk(relaxed = true)
+        every { messageBus.isDisposed } returns false
+        every { application.getService(SnykApplicationSettingsStateService::class.java) } returns mockk(relaxed = true)
+        every { application.getService(SnykAnalyticsService::class.java) } returns analyticsService
+        every { project.service<SnykCliDownloaderService>().isCliDownloading() } returns false
+    }
```

Comment:
with all those mocks seems for me as a good candidate for Integtest then...

Reply:
I already thought about using a LightPlatformTestCase as base to reduce mocks, as it gets a bit unwieldy.